sbi/inference/base.py

```diff
@@ -36,23 +38,33 @@ def __init__(
                  interpreted as a batch dimension but *currently* only the first batch
                  element will be used to condition on.
             simulation_batch_size: number of parameter sets that the
-                simulator accepts and converts to data x at once. If -1, we simulate 
-                all parameter sets at the same time. If >= 1, the simulator has to 
+                simulator accepts and converts to data x at once. If -1, we simulate
+                all parameter sets at the same time. If >= 1, the simulator has to
                 process data of shape (simulation_batch_size, parameter_dimension).
```

Comment:
Not related to this PR, but I suggest to rephrase this a bit. I find it confusing that the simulator "converts" theta to data, and that for -1 we simulate all parameter sets "at the same time". And I am confused by "parameter sets". So how about: 
```markdown
simulation_batch_size: Number of parameter vectors (batch size) that the simulator 
          function simulates per call. If the simulator can handle batches of parameters, 
          e.g., because it is vectorized or parallelized, this can give dramatic simulation speed ups. If 
          simulation_batch_size=-1, we pass all parameter vectors for a given round to the simulator at once. If 
          simulation_batch_size >= 1, we pass that number, e.g., the simulator is called with batches of shape 
          (simulation_batch_size, number of parameters in a single vector).
```

Reply:
In my previous PR (watch out for merge conflicts... @michaeldeistler ) I changed converts to maps to.

I also changed -1 to None. 

In general I approve of the spirit of @janfb's proposal - being super-clear. At least for me it is better now (but see the two minor comments above).